tests/metrics/README.md

```diff
@@ -12,19 +20,60 @@ To run the metrics, please use the Makefile rule `metrics-tests` on the top leve
 $ sudo -E PATH=$PATH make metrics-tests
 ```
 
-This will run all metrics tests and generete a `results` directory.
+This will run all metrics tests and place the results in the `lib/metrics/results` directory.
+
+Each file in the `results` directory contains the result for a test as a comma separated values (CSV).
 
-Each file in the `results` directory contains the result for a test as a comma separated values or CSV.
-At the end of each file you will find the result average of all the data collected by the test.
+### Run a single metrics test
+Some metrics tests require some files to be pre-processed before they can be executed. To pre-process
+these files execute:
+
+```bash
+$ make metrics-tests
+```
+Note, this command will also try to execute all the tests - you may wish to halt execution once the
+tests start running, as all the pre-processed files will have been generated at that point.
 
-Execute `sudo -E PATH=$PATH make metrics tests` to generate the necessary files to run individual tests.
-For example:
+To then run an indivudial test, you can for example:
 
 ```bash
 $ cd tests/metrics
 $ bash workload_time/cor_create_time.sh <times_to_run>
 ```
 
+Note: some tests require `root` privelege to execute correctly.
+
+## Internals
+This section covers some of the internal APIs and tools used by the metrics tests.
+
+### Results registration API
+The metrics tests register their results via the `save_results` function located in the common test
+library code. `save_results` takes four arguments:
+```
+save_results Name Args Result Units
+```
+
+| Argument | Description                                        |
+| -------- | -------------------------------------------------- |
+| Name     | The name of the test. This name is generally used to form a unique test identifer if the data is ultimately stored |
+| Args     | Any arugments that were passed to the test         |
+| Result   | The result of the test (without unit tags)         |
+| Units    | The Units the Result is measured in (KB, S etc.)   |
```

Comment:
OK, I've done some mods - let's see how the new one 'feels.
Still some confusion though - to note, there is no 'Result column' (did you mean 'row'?), and using 'Result' in the description for the Result row is somewhat self referential - it is a description of the content, 'result' is correct. I did change 'Units' to 'units' in the below Unit row though - as that one was wrong...